### ISSUE TITLE
Retain exception information - ?

### ISSUE THREAD

**Issue by cfcs** [mirage/mirage-platform#193]
Hey!

I'm using `mirage-xen`.

I frequently have the problem that exceptions raised here https://github.com/mirage/mirage-platform/blob/master/xen/lib/main.ml#L60 do not retain the original source in the backtrace.

Instead I get something like:
```
main: (Invalid_argument "String.sub / Bytes.sub")
Raised at file "src/core/lwt.ml" line 2719, characters 18-25
Called from file "lib/main.ml", line 60, characters 15-25
```

Does anybody know a way to obtain a more precise error location?

**Comment by hannesm** [comment]
I suspect it is the `lib/main.ml` in the xen subfolder of this repository. the main exception handler of MirageOS (on Xen) is implemented here: https://github.com/mirage/mirage-platform/blob/v3.0.4/xen/lib/main.ml#L50-L52

I've no clue which magic tricks we'd need to get "decent" backtraces with lwt...  not sure whether this is possible at all...

**Comment by hannesm** [comment]
I suspect this is a lwt issue rather than a mirage-xen one.  `Printexc.record_backtrace true` is always generated by `mirage` in `main.ml`, which enables backtraces.

**Comment by aantron** [comment]
If Lwt, it is likely [fixed in the recent 3.3.0](https://github.com/ocsigen/lwt/releases/tag/3.3.0).
